### PR TITLE
x.py setup: Create config.toml in the current directory, not the top-level directory

### PR DESCRIPTION
**Pull request by jyn514**
See https://github.com/rust-lang/rust/issues/78509 for discussion.

r? @pnkfelix
cc @cuviper @Mark-Simulacrum

### CODE REVIEW

File: src/bootstrap/setup.rs [hunk 1]
```diff
@@ -89,7 +89,7 @@ pub fn setup(src_path: &Path, profile: Profile) {
         std::process::exit(1);
     }
 
-    let path = cfg_file.unwrap_or_else(|| src_path.join("config.toml"));
+    let path = cfg_file.unwrap_or("config.toml".into());
```
Comment (lzutao) [inline src/bootstrap/setup.rs:92]:
```suggestion
    let path = cfg_file.unwrap_or_else(|| "config.toml".into());
```

Comment (jyn514) [reply src/bootstrap/setup.rs:92]: Performance doesn't matter for x.py. I'd rather have it be easier to read.

**Comment by Mark-Simulacrum** [comment]
r=me unless you specifically want to wait for @pnkfelix's review.

**Comment by jyn514** [comment]
@bors r=Mark-Simulacrum

I asked pnkfelix to review since I never heard back from them in the original thread, but from their bug report I think this is the behavior they expected.

**Pull request merged**